lib/web3/httpprovider.js

```diff
@@ -55,8 +55,16 @@ HttpProvider.prototype.sendAsync = function (payload, callback) {
     var request = new XMLHttpRequest();
     request.onreadystatechange = function() {
         if (request.readyState === 4) {
-            // TODO: handle the error properly here!!!
-            callback(null, JSON.parse(request.responseText));
+            var result = request.responseText;
+            var error = null;
+
+            try {
+                result = JSON.parse(result);
```

Comment:
This should also be add to the synchronous `send` function. (modified to work for sync and the errors should be thrown)